In [21]:
# input 4 digit number AND Reverse it
#Example 1
x=3456
rev=0
while(x>0):
  rev=(rev*10)+(x%10)
  x=x//10
print(int(rev))

6543


if we devide any no by 10 then it return remender will be last number<br/>
 3456%10 = 6<br/>
 rev=6<br/>
 2nd time we want last degit of 345 and skip the 6 for that we can devide it by 10 :<br/>
 x=3456/10=345.6<br/>
 int(345.6)=345<br/>
 now go to condition <br/>
 345>0 TRUE<br/>
 X%10 I.e 345%10=5<br/>
 but now we want to attach this after  6<br/>
 for that <br/>
 rev=rev*10+x%10<br/>
 rev was 6<br/>
 rev=(6*10)+(345%10)<br/>
    =60+5<br/>
 rev= 65<br/>
now we want 4 & skip 5<br/>
x//10<br/>
345//10 =34 // will skip . decimal places<br/>
34>0<br/>
rev was 65<br/>
rev=(65*10)+(34%10)= 650+4<br/>
rev=654<br/>
now we want last digit 3<br/>
so <br/>
x=34<br/>
34//10 =3<br/>
3>0<br/>
rev=654*10+(3%10)<br/>
rev= 6540+3<br/>
rev=6543<br/>
now <br/>
x=3<br/>
3//10=0.3<br/>
x>0<br/>
0.0> 0 false end while<br/>
print rev =6543<br/>